posts/2021-09-07-java17-210010-beta.adoc

```diff
@@ -0,0 +1,238 @@
+Although enabling this support may result in a performance improvement, it is recommended that the value for the `accessTokenCacheTimeout` attribute is short to reduce the possibility of a stale result as compared to what a validation call to the OpenID Connect Provider would have produced.
+
+You can also configure Liberty OIDC Social Login to process ID tokens that are in a `JWE` format. The corresponding OpenID Connect Provider should support creating `JWE` ID tokens.
+
+* Set the OpenID Connect Provider with the `OIDC` Social Login's public key that is used to encrypt the Content Encryption Key according to the OpenID Connect Provider's documentation.
+* Set the `keyManagementKeyAlias` attribute to the private key alias of the key management key that is used to decrypt the Content Encryption Key of `JWE` token. The key must exist in the `keyStore` configured for the SSL configuration referred by the the `sslRef` attribute. For example, `<oidcLogin keyManagementKeyAlias="privateKeyAlias" />`
+
+The signatureAlgorithm attributes of both elements now support the `RS384`, `RS512`, `HS384`, `HS512`, `ES256`, `ES384`, and `ES512` signature algorithms.
+
+`<openidConnectClient signatureAlgorithm="RS384"/>`
+`<oidcLogin signatureAlgorithm="RS384"/>`
+
+For more information about OpenID Connect Client refer to the link:https://openid.net/specs/openid-connect-core-1_0.html[OpenID Connect Client specification].
+
+[#mp_cp]
+=== MicroProfile Context Propagation 1.3
+
+MicroProfile Context Propagation 1.3 Release Candidate 1 adds support for Jakarta EE 9 packages (`jakarta.*`) and eventual alignment with the MicroProfile 5.0 platform.
+With the 1.3 RC1 release, you can now use MicroProfile Context Propagation within your Jakarta EE 9 applications.
+
+To enable the new beta features in your app, add them to your `server.xml`:
+
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>cdi-3.0</feature>
+    <feature>jndi-1.0</feature>
+    <feature>mpContextPropagation-1.3</feature>  
+    <feature>servlet-5.0</feature>
+  </featureManager>
+</server>
+----
+
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+
+[source, xml]
+----
+<dependency>
+    <groupId>org.eclipse.microprofile.context-propagation</groupId>
+    <artifactId>microprofile-context-propagation-api</artifactId>
+    <version>1.3-RC1</version>
+</dependency>
+----
+
+For more information on this MicroProfile Context Propagation update, refer to the link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/apidocs/[JavaDoc] and link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.3-RC1/microprofile-context-propagation-spec-1.3-RC1.html[specification].
+
+[#value_add]
+=== Jakarta EE 9 support for the remaining Open Liberty value-add features
+
+In Open Liberty 21.0.0.10-beta we now have Jakarta EE 9 support for the following value-add features:
```

Comment:
Change this section to the below:

=== Jakarta EE 9 support for the remaining Open Liberty value-add features

In January of this year we published link:{url-prefix}/blog/2021/01/26/ee9-messaging-security-21002-beta.html[Open Liberty 21.0.0.2-beta release] becoming first vendor product to be Jakarta EE Web Profile 9.0 compatible, and since then we've continued to deliver more Jakarta EE 9 support for our remaining features.  With the release of 21.0.0.10-beta, we're proud to announce that we've now completed our journey for adding Jakarta EE 9 support for all remaining value-add features!  They are as follows:

* Admin Center (link:{url-prefix}/docs/latest/reference/feature/adminCenter-1.0.html[adminCenter-1.0])
* Batch Management (link:{url-prefix}/docs/latest/reference/feature/batchManagement-1.0.html[batchManagement-1.0])
* gRPC (link:{url-prefix}/docs/latest/reference/feature/grpc-1.0.html[grpc-1.0])
* gRPC Client (link:{url-prefix}/docs/latest/reference/feature/grpcClient-1.0.html[grpcClient-1.0])
* Web Response Cache (link:{url-prefix}/docs/latest/reference/feature/webCache-1.0.html[webCache-1.0])
* Web Services Atomic Transaction (link:{url-prefix}/docs/latest/reference/feature/wsAtomicTransaction-1.2.html[wsAtomicTransaction-1.2])
* Web Services Security (link:{url-prefix}/docs/latest/reference/feature/wsSecurity-1.1.html[wsSecurity-1.1])
* Web Services Security SAML (link:{url-prefix}/docs/latest/reference/feature/wsSecuritySaml-1.1.html[wsSecuritySaml-1.1])

Existing server configurations that use these value-add features will now work with Jakarta EE 9 features when migrating applications in those server configuration to use Jakarta EE9 technologies.

Next on the horizon is MicroProfile 5.0, which will bring Jakarta EE 9 support to the various MicroProfile features.

Reply:
@jhanders34 does the above look good to you?

Reply:
I have some issues with the first paragraph.  Hopefully the below looks good to you.  We could add links to blog posts for things like `9.0 Full Platform support` and `9.1 Web Profile and Full Platform support`.

In January of this year we published the link:{url-prefix}/blog/2021/01/26/ee9-messaging-security-21002-beta.html[Open Liberty 21.0.0.2-beta release] blog post which announced Open Liberty becoming the first vendor product to be Jakarta EE Web Profile 9.0 compatible, and since then we've continued to deliver more Jakarta EE 9 support included 9.0 Full Platform support, 9.1 Web Profile and Full Platform support, and incremental support of Jakarta 9 for the Open Liberty value-add features. With the 21.0.0.10-beta release, we are proud to announce that we have now completed our journey by adding Jakarta EE 9 support for the remaining value-add features! The value-add features with Jakarta EE9 support that are added in this beta release are as follows: